db/db.go

```diff
@@ -445,6 +445,9 @@ func (d *DB) Reduce(events []core.Event) error {
 func defaultIndexFunc(d *DB) func(collection string, key ds.Key, oldData, newData []byte, txn ds.Txn) error {
 	return func(collection string, key ds.Key, oldData, newData []byte, txn ds.Txn) error {
 		c := d.GetCollection(collection)
+		if c == nil {
+			return fmt.Errorf("collection (%s) not found", collection)
```

Comment:
Keeping this msg short, because it is also going to be wrapped in the Dispatcher's error msg.